src/osd/scheduler/mClockScheduler.cc

```diff
@@ -86,15 +103,200 @@ const dmc::ClientInfo *mClockScheduler::ClientRegistry::get_info(
+  cct->_conf.set_val(
+    "osd_mclock_scheduler_background_recovery_lim", stringify(rec_lim));
+}
+
+void mClockScheduler::set_high_recovery_ops_profile_config()
+{
+  int rec_wgt = 10;
+  // Set client limit to 25% of max osd capacity
+  double client_lim = std::round(max_osd_capacity * 0.25);
+  // Set recovery limit to 75% of max osd capacity
+  double rec_lim = std::round(max_osd_capacity * 0.75);
+
+  // Set external client params
+  cct->_conf.set_val(
+    "osd_mclock_scheduler_client_res", stringify(default_min));
+  cct->_conf.set_val(
+    "osd_mclock_scheduler_client_wgt", stringify(default_min));
+  cct->_conf.set_val(
+    "osd_mclock_scheduler_client_lim", stringify(client_lim));
+
+  // Set background recovery client params
+  cct->_conf.set_val(
+    "osd_mclock_scheduler_background_recovery_res", stringify(default_min));
+  cct->_conf.set_val(
+    "osd_mclock_scheduler_background_recovery_wgt", stringify(rec_wgt));
+  cct->_conf.set_val(
+    "osd_mclock_scheduler_background_recovery_lim", stringify(rec_lim));
+}
+
+void mClockScheduler::set_high_client_ops_profile_config()
+{
+  int client_wgt = 10;
+  // Set client limit to 75% of max osd capacity
+  double client_lim = std::round(max_osd_capacity * 0.75);
+  // Set recovery limit to 25% of max osd capacity
+  double rec_lim = std::round(max_osd_capacity * 0.25);
+
+  // Set external client params
+  cct->_conf.set_val(
+    "osd_mclock_scheduler_client_res", stringify(default_min));
+  cct->_conf.set_val(
+    "osd_mclock_scheduler_client_wgt", stringify(client_wgt));
+  cct->_conf.set_val(
+    "osd_mclock_scheduler_client_lim", stringify(client_lim));
+
+  // Set background recovery client params
+  cct->_conf.set_val(
+    "osd_mclock_scheduler_background_recovery_res", stringify(default_min));
+  cct->_conf.set_val(
+    "osd_mclock_scheduler_background_recovery_wgt", stringify(default_min));
+  cct->_conf.set_val(
+    "osd_mclock_scheduler_background_recovery_lim", stringify(rec_lim));
+}
+
+int mClockScheduler::calc_scaled_cost(int cost)
+{
+  // Calculate bandwidth from max osd capacity (at 4KiB blocksize).
+  double max_osd_bandwidth = max_osd_capacity * 4 * 1024;
+
+  // Calculate scaled cost based on item cost
+  double scaled_cost = (cost / max_osd_bandwidth) * max_osd_capacity;
+
+  // Scale the cost down by an additional cost factor to
+  // account for different device characteristics (hdd, ssd).
+  scaled_cost *= osd_mclock_cost_per_io_msec / 1000.0;
+
+  return std::floor(scaled_cost);
+}
+
+bool mClockScheduler::maybe_update_client_cost_info(
+  op_type_t op_type, int new_cost)
+{
+  if (new_cost == 0) {
+    return false;
+  }
+
+  // The mclock params represented in terms of the per-osd capacity
+  // are scaled up or down according to the cost associated with
+  // item cost and updated within the dmclock server.
+  int cur_cost = client_cost_infos[op_type];
+  bool cost_changed =
+    (((new_cost << 1) >= cur_cost) ||
```

Comment:
nit: the condition is already a bool, there's no need for the ternary operator

Reply:
Yes, I had a different calculation before and ternary operator got retained. Will remove it. Perils of working late nights..<sigh>.